lib/dropbox_api/client.rb

```diff
@@ -1,7 +1,11 @@
 module DropboxApi
   class Client
     def initialize(oauth_bearer = ENV["DROPBOX_OAUTH_BEARER"])
-      @connection_builder = ConnectionBuilder.new(oauth_bearer)
+      @connection_builder = ConnectionBuilder.new(oauth_bearer, middleware: middleware)
+    end
+
+    def middleware
+      @middleware ||= MiddleWare::Stack.new
```

Comment:
It's not very clear if the middleware object lives in `Client` or `ConnectionBuilder`. I can see it's initialized in `Client` and passed over to `ConnectionBuilder`, so in practice it lives in `Client` but `ConnectionBuilder` has a reference.

Would it make sense to have it defined in `ConnectionBuilder`? Then we can expose it through `Client` by just delegating it.

Reply:
Yeah, I think that's a good idea!